### PR TITLE
Remove context from structs

### PR DESCRIPTION
**Pull request by orestisfl**
### Summary of your changes
Contexts in structs are not a good practice in Go: https://go.dev/blog/context-and-structs

Even though this shouldn't affect the current code, we should make sure we avoid this practice so that we don't run in problems in the future.

Enable the [containedctx](https://github.com/sivchari/containedctx) linter.

`posture` and `Manager` are exceptions since one is called by the libbeat API and the other needs it for goroutine canceling (code could be refactored there as well but is a bigger change).

I propose to not immediately merge this as it might introduce conflicts for the 8.11 branch during the QA cycle.

### Related Issues
- Related: https://github.com/elastic/cloudbeat/issues/691
- Related: https://github.com/elastic/cloudbeat/issues/1108
- Related: https://github.com/elastic/cloudbeat/issues/1218

### Checklist
- [X] I have added tests that prove my fix is effective or that my feature works
- [X] I have added the necessary README/documentation (if appropriate)

### CODE REVIEW

File: resources/providers/awslib/ec2/provider_test.go [hunk 1]
```diff
@@ -270,119 +268,6 @@ func TestProvider_GetEbsEncryptionByDefault(t *testing.T) {
 	}
 }
 
-func TestProvider_DescribeInstances(t *testing.T) {
```
Comment (orestisfl) [inline resources/providers/awslib/ec2/provider_test.go:273]: Removed these tests as they are misleadingly empty

File: resources/providers/azurelib/inventory/provider_test.go [hunk 2]
```diff
@@ -145,29 +142,16 @@ func (s *ProviderTestSuite) TestGetString() {
 	}
 }
 
-func (s *ProviderTestSuite) TestProviderInit() {
```
Comment (orestisfl) [inline resources/providers/azurelib/inventory/provider_test.go:148]: Test only checks mock

File: resources/providers/gcplib/inventory/provider_test.go [hunk 3]
```diff
@@ -68,8 +67,8 @@ func (s *ProviderTestSuite) TestProviderInit() {
 		ClientOpts: []option.ClientOption{},
 	}
 
-	initMock.On("Init", s.ctx, s.logger, gcpConfig).Return(&Provider{}, nil).Once()
-	provider, err := initMock.Init(s.ctx, s.logger, gcpConfig)
+	initMock.On("Init", mock.Anything, s.logger, gcpConfig).Return(&Provider{}, nil).Once()
```
Comment (amirbenun) [inline resources/providers/gcplib/inventory/provider_test.go:70]: `EXPECT`

**Review by amirbenun — APPROVED** [review]
💯

**Pull request merged**